src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +135,1023 @@ static DamageTable D_80A63CE0 = {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62FAC.s")
+void EnBal_TryBalloonPopped(EnBal* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->picto.actor.world.pos.x,
+                    this->picto.actor.world.pos.y + 100.0f, this->picto.actor.world.pos.z, 0xFF, 0xFF, 0xC8, 2);
+        this->collider.base.acFlags &= ~AC_HIT;
+        EnBal_SetupPopBalloon(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A63158.s")
+void EnBal_Update(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A634B4.s")
+    this->actionFunc(this, play);
+    EnBal_TryBalloonPopped(this, play);
+    EnBal_UpdateCollision(this, play);
+    SkelAnime_Update(&this->skelAnime);
+    if (this->actionFunc != EnBal_FloatIdle) {
+        Actor_MoveWithGravity(&this->picto.actor);
+        Actor_UpdateBgCheckInfo(play, &this->picto.actor, 32.0f, 30.0f, 60.0f,
+                                UPDBGCHECKINFO_FLAG_2 | UPDBGCHECKINFO_FLAG_4);
+    }
+    EnBal_UpdateHead(this, play);
+    if ((this->actionFunc == EnBal_Talk) && (this->textId != 0x1D10)) {
+        Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+        this->picto.actor.world.rot.y = this->picto.actor.shape.rot.y;
+    }
+    EnBal_UpdateShadow(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A634C8.s")
+s32 EnBal_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnBal* this = THIS;
+    Vec3s balloonRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A635DC.s")
+    if (limbIndex == TINGLE_LIMB_BALLOON) {
+        if ((this->balloonAction == TINGLE_BALLOON_ACTION_NONE) ||
+            (this->balloonAction == TINGLE_BALLOON_ACTION_INFLATE) ||
+            (this->balloonAction == TINGLE_BALLOON_ACTION_RISE)) {
+            balloonRot.x = Math_SinS(this->timer) * 0xE38;
+            balloonRot.z = Math_CosS(this->timer) * 0xE38;
+            Matrix_RotateZYX(balloonRot.x, 0, balloonRot.z, MTXMODE_APPLY);
+            Matrix_Scale(this->balloonScale.x, this->balloonScale.y, this->balloonScale.z, MTXMODE_APPLY);
+            Matrix_RotateZS(-balloonRot.z, MTXMODE_APPLY);
+            Matrix_RotateXS(-balloonRot.x, MTXMODE_APPLY);
+        } else {
+            *dList = NULL;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A635F0.s")
+    if (limbIndex == TINGLE_LIMB_HEAD) {
+        rot->x += this->headRot.y;
+        rot->y -= this->headRot.x;
+        if ((this->isTalking == true) && !(play->state.frames & 2)) {
```

Comment:
```suggestion
        if ((this->isTalking == true) && ((play->state.frames & 2) == 0)) {
```
Taking the `&` of the frames as a check is a bit different than a bool since it often includes a range of values that isn't 1 bit (it is 1 bit in this case but we keep it `== 0` for more general cases)